webapp/channels/src/components/admin_console/ip_filtering/index.tsx

```diff
@@ -27,16 +30,36 @@ import SaveChangesPanel from '../team_channel_settings/save_changes_panel';
 import './ip_filtering.scss';
 
 const IPFiltering = () => {
-    const dispatch = useDispatch();
+    const dispatch = useDispatch<DispatchFunc>();
     const {formatMessage} = useIntl();
     const [ipFilters, setIpFilters] = useState<AllowedIPRange[] | null>(null);
     const [originalIpFilters, setOriginalIpFilters] = useState<AllowedIPRange[] | null>(null);
     const [saveNeeded, setSaveNeeded] = useState(false);
     const [currentUsersIP, setCurrentUsersIP] = useState<string | null>(null);
     const [saving, setSaving] = useState<boolean>(false);
     const [filterToggle, setFilterToggle] = useState<boolean>(false);
+    const [installationStatus, setInstallationStatus] = useState<string>('');
+
+    // savingMessage allows the component to change the label on the Save button in the SaveChangesPanel
+    const [savingMessage, setSavingMessage] = useState<string>('');
+
+    // savingDescription is a JSX element that will be displayed in the serverError bar on the SaveChangesPanel. This allows us to provide more information on loading while previous changes are applied
+    const [savingDescription, setSavingDescription] = useState<JSX.Element | null>(null);
+    const [installationFetchAttempts, setInstallationFetchAttempts] = useState<number>(0);
```

Comment:
If we get rid of the effect and move the logic to the callback, we can use a ref instead of a state.

Reply:
I was able to eliminate this entirely in the refactor [here](https://github.com/mattermost/mattermost/pull/25485#discussion_r1406476412)